nshlib/nsh_console.h

```diff
@@ -157,6 +157,14 @@ struct nsh_vtbl_s
   char traceline[CONFIG_NSH_LINELEN];
 #endif
 
+  /* Temporary line buffer */
+
+#if (!defined(CONFIG_NSH_DISABLE_MEMDUMP) && defined(NSH_HAVE_WRITEFILE)) || \
+    !defined(CONFIG_NSH_DISABLEBG) || defined(CONFIG_NSH_PIPELINE) || \
+    !defined(CONFIG_NSH_DISABLE_WATCH)
+  char templine[CONFIG_NSH_LINELEN];
```

Comment:
but this will increase the heap consumption for the temp use case.

Reply:
Compared with irregular heap access, shared template and will make it more deterministic

Reply:
could you split the rename from moving buffer to nsh_vtbl?

Reply:
the temp buffer is only used in few command, why move it to nsh_vtbl to extend the life to the whole session?